### PR TITLE
Final and working product merging into main.

### PR DESCRIPTION
**Pull request by CraftedByMatt**
This PR aims to move the entirety of the project to the main branch as the final product is complete.

### CODE REVIEW

File: spec/factories/preferences.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#  restriction :boolean          default(FALSE), not null
+#  user_id     :bigint           not null
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+# Indexes
+#
+#  index_preferences_on_user_id  (user_id)
+#
+FactoryBot.define do
```
Comment (brunoparma88) [inline spec/factories/preferences.rb:17]: use Faker

File: spec/features/preferences/edit_preference_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    sign_in user
+    visit preferences_path
+  end
+
+  describe '#update' do
+    context 'with valid attributes' do
+      it 'updates the requested preference' do
+        edit_first_preference('New preference')
+        expect(page).to have_content('Preference successfully updated.')
+        expect(find('input[name="preference[name]"]').value).to eq('New preference') # Css Selector was needed to target field.
```
Comment (brunoparma88) [inline spec/features/preferences/edit_preference_spec.rb:19]: Remove unnecessary comment

File: db/schema.rb [hunk 3]
```diff
@@ -103,12 +103,12 @@
   end
 
   create_table "recipes", force: :cascade do |t|
-    t.string "name", null: false
-    t.text "description", null: false
+    t.string "name"
```
Comment (brunoparma88) [inline db/schema.rb:106]: create_table "recipes" did it already exist?

**Comment by CraftedByMatt** [comment]
We are disregarding this PR and moving over to https://github.com/CraftedByMatt/learningcamp-template/pull/10 for the merge of the whole project as well as fixes from the comments.

**Pull request closed without merging**